meteor/client/ui/SegmentTimeline/Parts/SegmentTimelinePart.tsx

```diff
@@ -80,6 +80,10 @@ interface IProps {
 	cropDuration?: number
 	className?: string
 	showDurationSourceLayers?: Set<ISourceLayer['_id']>
+	isLiveSegment?: boolean
```

Comment:
I think some candidates that will generally change together/not change very often are `firstPartInSegment`, `lastPartInSegment`, `isLastInSegment`, `isLastSegment`.

Reply:
> I am not a React expert, but from what know grouping could cause more renders than necessary because React only shallow compares properties when checking if the props have changed. It is possible to either force react to do deep diffing instead, or to memoize the objects before passing them in, but that is more work than a simple rearranging of properties. @jstarpl or do you think this is not something worth worrying about?

Well, yeah, if we go with object props then they will need to be memoized, especially in a very render-intensive component like this one - and I think, not explicitly because of the comparison problem (this particular component already does a deep equals for `shouldComponentUpdate`), but because GC'ing all of these objects created at 60 fps will cause hickups.